apps/e2e/cypress/e2e/talentsearch/submit-application-iap.cy.js

```diff
@@ -0,0 +1,338 @@
+    });
+    cy.visit("/en/indigenous-it-apprentice");
+
+    // IAP home page
+    cy.wait("@gqlIAPPublishedPoolsQuery");
+    cy.findByRole("heading", {
+      name: /IT Apprenticeship Program for Indigenous Peoples/i,
+      level: 1,
+    })
+      .should("exist")
+      .and("be.visible");
+    cy.get("@testPoolId").then((testPoolId) => {
+      cy.get(
+        `a[href*="/browse/pools/${testPoolId}/create-application?personality=iap"]`,
+      )
+        .first() // there are several buttons on the page to "Apply Now"
+        .click();
+    });
+    cy.wait("@gqlcreateApplicationMutation");
+
+    // Welcome page - step one
+    cy.wait("@gqlGetApplicationQuery");
+    cy.findByRole("heading", {
+      name: "Apply to Cypress Test Pool EN",
+      level: 1,
+    })
+      .should("exist")
+      .and("be.visible"); // IAP heading does not include job stream
+    cy.findByRole("heading", {
+      name: /Welcome, Cypress/i,
+    })
+      .should("exist")
+      .and("be.visible");
+    cy.findByRole("heading", { name: /Step 1 of 7/i }) // IAP workflow has one extra step, but we're removing the screening questions step
+      .should("exist")
+      .and("be.visible");
+    cy.contains(
+      "The program is a Government of Canada initiative specifically for First Nations, Inuit, and Métis peoples.",
+    ); // customized copy for IAP
+    cy.findByRole("button", { name: /Let's go!/i }).click();
+    cy.wait("@gqlUpdateApplicationMutation");
+
+    // Self-Declaration - step two
+    cy.findByRole("heading", {
+      name: /Indigenous Peoples Self-Declaration Form/i,
+    })
+      .should("exist")
+      .and("be.visible");
+    cy.findByRole("heading", { name: /Step 2 of 7/i }) // self declaration is the new step 2
+      .should("exist")
+      .and("be.visible");
+
+    cy.findByRole("radio", {
+      // don't affirm
+      name: /I am not a member of an Indigenous group/i,
+    }).click();
+    cy.findByRole("button", { name: /Sign and continue/i }).should("not.exist"); // check that you cannot proceed without affirmation
+
+    cy.findByRole("radio", {
+      // affirmation
+      name: /I am First Nations/i,
+    }).click();
+    cy.findByRole("button", { name: /Sign and continue/i }).should("not.exist"); // check that you cannot proceed without community selection
+
+    cy.findByRole("checkbox", {
+      // community selection
+      name: /I am First Nations/i,
+    }).click();
+    cy.findByRole("radio", {
+      // community sub-selection
+      name: /I am Status First Nations/i,
+    }).click();
+    cy.findByRole("textbox", {
+      name: /signature/i,
+    }).type(uniqueTestId);
+    cy.findByRole("button", { name: /Sign and continue/i }).click();
+    cy.wait("@gqlUpdateUserAndApplicationMutation");
+
+    // Review profile page - step three
+    cy.wait("@gqlGetApplicationQuery");
+    cy.findByRole("heading", { name: /Review your profile/i })
+      .should("exist")
+      .and("be.visible");
```

Comment:
I got further 😅 Looks like this one depends on the `getMe` query so we should also wait on that.

Reply:
Hmm, I'm a bit confused now.  Is the regular submit-application.cy.js failing for you as well?  What is happening, exactly?  Is the `findByRole` timing out after x seconds while the page slowly loads?

Reply:
Yeah, the `findByRole` times out because the `getMe` query is still resolving and the heading is actually not rendered until that query completes. 

I will check the other spec and let you know 👍

Reply:
OK, I removed the waits and bumped the command timeout way up.  I think this is more realistic to the way a user would interact with the site, instead of waiting for each Graphql request one-by-one.